spec/controllers/prompts_controller_spec.rb

```diff
@@ -0,0 +1,131 @@
+  include LoginMacros
+
+  before do
+    fake_login
+  end
+
+  describe 'no_challenge' do
+    it 'should show an error and redirect' do
+      collection = create(:collection)
+      get :show, collection_id: collection.name
+      expect(response).to redirect_to(collection_path(collection))
+      expect(flash[:error]).to eq "What challenge did you want to sign up for?"
+    end
+  end
+
+  describe 'no_signup' do
+    it 'should show an error and redirect' do
+      signups = create(:challenge_signup)
+      post :create, collection_id: signups.collection.name
+      expect(response).to redirect_to(collection_signups_path(signups.collection)+'/new')
+      expect(flash[:error]).to eq "Please submit a basic sign-up with the required fields first."
+    end 
+  end
+
+  describe 'signups_closed' do
+    it 'should show an error and redirect' do
+      signups = create(:challenge_signup)
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      #signups.collection.collection_preference.closed = true
+      post :create, collection_id: signups.collection.name 
+      expect(response).to redirect_to(collection_path(signups.collection))
+      expect(flash[:error]).to eq "Signup is currently closed: please contact a moderator for help."
+    end
+ end
+
+  describe 'not_signup_owner' do
+    it 'should show an error and redirect' do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      post :edit, id: prompt.id, collection_id: signups.collection.name
+      expect(response).to redirect_to(collection_path(signups.collection))
+      expect(flash[:error]).to eq "You can't edit someone else's sign-up!"
+    end
+  end
+
+  describe 'new_prompt_offer' do
+    it 'should have no errors and redirect' do
+      signups = create(:challenge_signup)
+      signups.collection.challenge.signup_open = true
+      signups.collection.challenge.save
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      post :new, collection_id: signups.collection.name, prompt_type: 'offer'
+      expect(response).to have_http_status(200)
+      expect(flash[:error]).blank?
+    end
+  end
+
+  describe 'create_prompt_offer' do
+    it 'should have no errors and redirect' do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      signups.collection.challenge.signup_open = true
+      signups.collection.challenge.save
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      post :create, collection_id: signups.collection.name, prompt_type: 'offer', prompt: {}
+      expect(response).to redirect_to("#{collection_signups_path(signups.collection)}/#{prompt.challenge_signup_id}/edit")
```

Comment:
Hm -- I think I'm confused. If this is testing `create`, we only redirect to the edit signup path when the prompt is invalid... but if that's the case, we'd `expect(flash[:error]).to eq "That prompt would make your overall sign-up invalid, sorry."`. What are you testing here? A `context` would be helpful.